ts/Core/Axis/Axis.ts

```diff
@@ -1833,12 +1833,24 @@ class Axis {
         // This is in turn needed in order to find tick positions in ordinal
         // axes.
         if (isXAxis && !secondPass) {
+            // First process all series assigned to that axis.
             axis.series.forEach(function (series): void {
+                const chart = series.chart,
+                    options = series.options,
+                    dataGroupingOptions = options.dataGrouping,
+                    groupingEnabled = series.allowDG !== false && dataGroupingOptions &&
+                        pick(dataGroupingOptions.enabled, chart.options.isStock);
+
+                // Allows filtering out points outside the plot area.
+                series.forceCrop = groupingEnabled; // #334
```

Comment:
Do we need this in Axis core? Stock related code should be in the Stock modules

Reply:
Inside that loop the `series.processData()` is triggered. Then this method fires the `getProcessedData()`.
If I don't add this line (`series.forceCrop = groupingEnabled`) the method (`getProcessedData`) returns uncropped data that causes some further issues.
At that point, I should know if the data should be cropped or not.

I could an event like for example `beforeProcessData` because I guess we shouldn't play with `cropThreshold` here?

Reply:
Perhaps we should replace the flag `forceCrop` with a method that returns this info? Something like this:

```js
forceCropping () {
  return this.forceCrop || (
    this.allowDG !== false &&
    dataGroupingOptions &&
    pick(dataGroupingOptions.enabled, this.chart.options.isStock);
  );
}
``` 

And use it in `getProcessedData()`? What do you think @TorsteinHonsi ?

Just to explain the solution we are trying to achieve here:
Currently we process and group data in this way (pseudo-code):
```js
axis.series.forEach(series => {
  series.processData();
  if (groupingNeeded) { series.groupData(); }
});
```

This way first series might not be grouped but the second one will be. That causes problems with stacking (different x-values for stacks).

Instead, we decided to solve it this way:
```js
axis.series.forEach(series => {
  series.processData();
});
// All series processed, we know groupPixelWidth for the most dense series
axis.series.forEach(series => {
  if (groupingNeeded) { series.groupData(); }
});
```

Reply:
Maybe I am missing something here, but we already have logic that is supposed to go over all the series and identify a common `groupPixelWidth`:

https://github.com/highcharts/highcharts/blob/627e742c23721b32e2c744cee92a5a04659e41b5/ts/Extensions/DataGrouping.ts#L1163-L1165

Reply:
We do have the `getGroupPixelWidth` method, but in order to get the `groupPixelWidth` the data needs to be processed. And in this case, we have to flag that **before** processing.

``` TS
// If one of the series needs grouping, apply it to all (#1634)
    i = len;
    while (i--) {
        dgOptions = series[i].options.dataGrouping;

        if (dgOptions) { // #2692

            dataLength = (series[i].processedXData || series[i].data).length;

            // Execute grouping if the amount of points is greater than the
            // limit defined in groupPixelWidth
            if (
                series[i].groupPixelWidth ||
                dataLength >
                ((this.chart.plotSizeX as any) / groupPixelWidth) ||
                (dataLength && dgOptions.forced)
            ) {
                doGrouping = true;
            }
        }
    }
```

The `forceCrop` flag rather checks if the grouping might be needed in the future. 

If you take a look at the `getProcessedData` the part that is failing to crop is the one below. I don't think that we can add any other property to make it pass.

``` TS
        // optionally filter out points outside the plot area
        if (isCartesian &&
            series.sorted &&
            !getExtremesFromAll &&
            (
                !cropThreshold ||
                dataLength > cropThreshold ||
                series.forceCrop
            )
        ) {
```

I will suggest adding a function like Pawel suggested (`forceCropping`)